prqlc/prql-compiler/tests/integration/queries.rs

```diff
@@ -8,123 +8,145 @@ use prql_compiler::sql::Dialect;
+    fn run(prql_path: &Path) {
+        let test_name = prql_path.file_stem().unwrap().to_str().unwrap();
+        let prql = fs::read_to_string(prql_path).unwrap();
 
-        let is_contains_data_root = re.is_match(&prql);
+        let data_file_root_keyword = "data_file_root";
+        let is_contains_data_root = prql.contains(data_file_root_keyword);
 
-        // for each of the dialects
-        insta::allow_duplicates! {
-        for con in &mut connections {
-            if !con.dialect.should_run_query(&prql) {
+        // for each of the connections
+        let mut results = Vec::new();
+        for con in CONNECTIONS.lock().unwrap().deref_mut() {
+            if !con.should_run_query(&prql) {
                 continue;
             }
-            let dialect = con.dialect;
-            let options = Options::default().with_target(Sql(Some(dialect)));
+
             let mut prql = prql.clone();
             if is_contains_data_root {
-                prql = re.replace_all(&prql, &con.data_file_root).to_string();
+                prql = prql.replace(data_file_root_keyword, &con.cfg.data_file_root);
             }
-            let mut rows = prql_compiler::compile(&prql, &options)
-                .and_then(|sql| Ok(con.protocol.run_query(sql.as_str(), runtime)?))
+
+            let dialect = con.cfg.dialect;
+
+            let rows = con
+                .run_query(&prql)
                 .context(format!("Executing {test_name} for {dialect}"))
                 .unwrap();
 
-            // TODO: I think these could possibly be moved to the DbConnection impls
-            crate::dbs::replace_booleans(&mut rows);
-            crate::dbs::remove_trailing_zeros(&mut rows);
-
+            // convert into ad-hoc CSV
             let result = rows
                 .iter()
-                // Make a CSV so it's easier to compare
-                .map(|r| r.iter().join(","))
+                .map(|r| r.join(","))
+                .collect::<Vec<_>>()
                 .join("\n");
 
-            // Add message so we know which dialect fails.
-            insta::with_settings!({
-                description=>format!("# Running on dialect `{}`\n\n# Query:\n---\n{}", &con.dialect, &prql)
-            }, {
-                assert_snapshot!("results", &result, &prql);
-            })
+            results.push((dialect, result));
         }
+
+        // insta::allow_duplicates!, but with reporting of which two cases are not matching.
+        let (left_dialect, left_text) = results.swap_remove(0);
+        for (right_dialect, right_text) in results {
+            similar_asserts::assert_eq!(
+                left_text,
+                right_text,
+                "{} {} {}",
+                test_name,
+                left_dialect,
+                right_dialect
+            );
         }
-    })
+
+        with_settings!({ input_file => prql_path }, {
+            assert_snapshot!(test_name, left_text, &prql)
```

Comment:
`description` isn't that helpful given it doesn't seem to show on failures within `glob!`, though is worth _removing_? (from https://github.com/PRQL/prql/pull/3871/files#diff-194853534e9688ae9ff624c9936b82108b3fa8aea5d5bb02dc503047395b2ef4L123)

(and possibly now we're not using `glob!`, it does show?)